src/ucp/rndv/rndv.c

```diff
@@ -817,20 +906,17 @@ ucp_rndv_recv_frag_put_mem_type(ucp_request_t *rreq, ucp_request_t *rndv_req,
      * frag recv buffer to memtype recv buffer
      */
 
-    ucp_rndv_init_mem_type_frag_req(rreq->recv.worker, freq, UCP_REQUEST_SEND_PROTO_RNDV_PUT,
-                                    ucp_rndv_recv_frag_put_completion, mdesc, rreq->recv.mem_type,
-                                    length, ucp_rndv_progress_rma_put_zcopy);
+    ucp_rndv_init_mem_type_frag_req(rreq->recv.worker, freq,
+                                    UCP_REQUEST_SEND_PROTO_RNDV_PUT,
+                                    ucp_rndv_recv_frag_put_completion, mdesc,
+                                    rreq->recv.mem_type, length,
+                                    ucp_rndv_progress_rma_put_zcopy);
 
-    freq->super_req                    = rreq;
```

Comment:
is this set somewhere else? it is used in ucp_rndv_recv_frag_put_completion

Reply:
now we don't overwrite `freq->super_req` by `rreq`. So, in case of GET protocol, it is `freq` -> `rndv_req` -> `rreq`. For PUT protocol - `freq` -> `rreq`
so, we don't need to allocate request ID for `rndv_req` to save it